app/models/vm_or_template.rb

```diff
@@ -356,6 +356,27 @@ def run_command_via_parent(verb, options = {})
     ext_management_system.send(verb, self, options)
   end
 
+  # keep the same method signature as others in retirement mixin
+  def self.make_retire_request(*src_ids, requester, initiated_by: 'user')
+    src_ids.each do |src_id|
+      target = find_by(:id => src_id)
+      if target.nil?
+        _log.error("Retirement of [Vm] ID: [#{src_id}] skipped - target does not exist")
+        next
+      end
+
+      target.check_policy_prevent('request_vm_retire', "retire_request_after_policy_check", src_id, requester.userid, :initiated_by => initiated_by)
+    end
+  end
+
+  def retire_request_after_policy_check(*src_ids, userid, initiated_by: 'user')
+    klass = VmRetireRequest
```

Comment:
Is there a reason to put the class into a variable if it never modified?

Can we just call `VmRetireRequest.request_types.first` and `VmRetireRequest.make_request` directly in this method?